File: MainCore/server/godserver.py
from flask import Flask, request, jsonify
from flask_cors import CORS
from pymongo import MongoClient
from werkzeug.security import generate_password_hash, check_password_hash



app = Flask(__name__)
CORS(app)

client = MongoClient("mongodb+srv://petwhiz8:VNs3DXlFys93zAzE@cluster1.tdxh1k7.mongodb.net/?retryWrites=true&w=majority")
db = client["clientdatas"]
collection = db["userinfo"]

@app.route('/register', methods=['POST'])
def register():
    data = request.json
    print("request:", data)
    username = data.get('username')
    email = data.get('email')
    password = data.get('password')

    if not username or not email or not password:
        return jsonify({'error': 'Incomplete data provided'}), 400


    existing_user = collection.find_one({'$or': [{'username': username}, {'email': email}]})
    if existing_user:
        return jsonify({'error': 'Username or email is already taken'}), 400


    hashed_password = generate_password_hash(password)

    new_user = {
        'username': username,
        'email': email,
        'password': hashed_password
    }
    collection.insert_one(new_user)

    return jsonify({'message': 'User registered successfully'}), 201

@app.route('/login', methods=['POST'])
def login():
    data = request.json
    username = data.get('username')
    password = data.get('password')
    user = collection.find_one({'username': username})
    if user and check_password_hash(user['password'], password):
        return jsonify({'success': True, 'message': 'Login successful'}), 200
    else:
        return jsonify({'success': False, 'message': 'Invalid username or password'}), 401
if __name__ == '__main__':
    app.run(debug=True, port=5007)

File: MainCore/server/godserver.js
const express = require('express');
const mongoose = require('mongoose');
const bodyParser = require('body-parser');
const cors = require('cors');

const app = express();
const PORT = process.env.PORT || 3001;

// MongoDB connection string
const mongoURI = "mongodb+srv://petwhiz8:VNs3DXlFys93zAzE@cluster1.tdxh1k7.mongodb.net/?retryWrites=true&w=majority";

// Connect to MongoDB
mongoose.connect(mongoURI)
    .then(() => console.log('Connected to MongoDB'))
    .catch(err => console.error('Could not connect to MongoDB', err));

// Define a schema for the user collection
const userSchema = new mongoose.Schema({
    username: String,
    email: String,
    password: String
});

// Create a model based on the schema
const User = mongoose.model('User', userSchema, 'usersdata');

// Middleware to parse JSON request bodies
app.use(bodyParser.json());

// Enable CORS for all origins
app.use(cors());


// Route to handle user registration
app.post('/register', async (req, res) => {
    const { username, email, password } = req.body;

    try {
        // Check if the email is already registered
        const existingUser = await User.findOne({ username });
        if (existingUser) {
            return res.status(400).json({ error: 'Email is already registered' });
        }

        // Create a new user instance
        const newUser = new User({
            username,
            email,
            password
        });

        // Save the user to the database
        await newUser.save();

        // Retrieve the user from the database to ensure all fields are included
        const registeredUser = await User.findById(newUser._id);

        res.status(201).json({ message: 'User registered successfully', user: registeredUser });
    } catch (error) {
        console.error('Error registering user:', error);
        res.status(500).json({ error: 'Internal server error' });
    }
});

// Route to handle user login
app.post('/login', async (req, res) => {
    const { username, password } = req.body;

    try {
        // Find the user by username
        const user = await User.findOne({ username });
        if (!user) {
            return res.status(401).json({ error: 'Invalid username or password' });
        }

        // Check if the password matches
        if (user.password !== password) {
            return res.status(401).json({ error: 'Invalid username or password' });
        }

        // Successfully logged in
        res.status(200).json({ message: 'Login successful' });
    } catch (error) {
        console.error('Error logging in user:', error);
        res.status(500).json({ error: 'Internal server error' });
    }
});


// Start the server
app.listen(PORT, () => {
    console.log(`Server is running on port ${PORT}`);
});
